### PR TITLE
fix(pswap-distribution): add pool and liquidity of TBCD for benchmark

### LINKED ISSUE #849: Fix pswap-distribution benchmark
pswap-distribution benchmark (not test) panics.

How to reproduce:
```
cargo run --bin framenode --features private-net,runtime-benchmarks,ready-to-test benchmark pallet --chain=local  --execution=native --wasm-execution=compiled --pallet pswap_distribution --extrinsic "*" --steps 2 --repeat 1 --output  ./benches
```

Error:
```
Thread 'main' panicked at 'assertion failed: Assets::<T>::free_balance(&PSWAP.into(), &liquidity_provider).unwrap() >\n    balance!(0)', /home/wer/workspace/sora2-network/pallets/pswap-distribution/benchmarking/src/lib.rs:199

2023-12-14 16:34:21 Evicting failed runtime instance error=Runtime panicked: assertion failed: Assets::<T>::free_balance(&PSWAP.into(), &liquidity_provider).unwrap() >\n    balance!(0)
Error: Input("Error executing and verifying runtime benchmark: Runtime panicked: assertion failed: Assets::<T>::free_balance(&PSWAP.into(), &liquidity_provider).unwrap() >\\n    balance!(0)")
```

### PR DESCRIPTION
**Pull request by i3ima**
Fixes #849. Not sure if my fix is correct as it's my first task after onboarding. Also I'm not sure if feature flags should be changed after this one because benchmarks are running with `--features private-net,runtime-benchmarks,ready-to-test`

### CODE REVIEW

File: pallets/dex-api/src/lib.rs [hunk 1]
```diff
@@ -331,21 +331,19 @@ impl<T: Config>
...
+                ) =>
                 {
                     Some(LiquiditySourceId::new(
                         filter.dex_id.clone(),
                         source_type.clone(),
                     ))
-                } else {
-                    None
                 }
+                _ => None,
```
Comment (wer1st) [inline pallets/dex-api/src/lib.rs:346]: For what? `if-else` looks better for bool than `match`.
Looks redundant, please revert

Comment (i3ima) [reply pallets/dex-api/src/lib.rs]: It's just that I'm a fan of such expressions. Reverted.

**Review by wer1st — COMMENTED** [review]
Benches are run by 2 ways:
- As unit tests
- As benchmarking process

Please test both scenarios:
To run as unit test

**Review by wer1st — APPROVED** [review]

**Comment by i3ima** [comment]
> Benches are run by 2 ways:
> 
> * As unit tests
> * As benchmarking process
> 
> Please test both scenarios: To run as unit test:
> 
> ```shell
> cargo test --features private-net,wip,ready-to-test,runtime-benchmarks
> ```
> 
> To run as benchmarks:
> 
> ```shell
> mkdir benches
> cargo run --release --bin framenode --features private-net,runtime-benchmarks benchmark pallet --chain=local --execution=wasm --wasm-execution=compiled --pallet pswap-distribution --extrinsic "*" --steps 50 --repeat 20 --output  ./benches
> ```

All passes.

Tests: 
```shell
❯ cargo test --features private-net,wip,ready-to-test,runtime-benchmarks
# Bunch of test lines...
   Doc-tests xst-benchmarking

running 0 tests

test result: ok. 0 passed; 0 failed; 0 ignored; 0 measured; 0 filtered out; finished in 0.00s

❯ echo $?
0
```

Benchmarks:
```shell
❯ cargo run --release --bin framenode --features private-net,runtime-benchmarks benchmark pallet --chain=local --execution=wasm --wasm-execution=compiled --pallet pswap-distribution --extrinsic "*" --steps 50 --repeat 20 --output  ./benches
# Benchmark logs...

Quality and confidence:
param     error
a         0.151
b         0.774
c         0.081

Model:
Time ~=    339.7
    + a    4.746
    + b    273.7
    + c    6.805
              µs

Reads = 123 + (1 * a) + (0 * b) + (1 * c)
Writes = 91 + (0 * a) + (0 * b) + (0 * c)
Recorded proof Size = 4889 + (248 * a) + (947 * b) + (50 * c)

Created file: "./benches/pswap_distribution.rs"
❯ echo $?
0
```

By the way I thought one of CIs runs such scenarios  🤷.

**Comment by wer1st** [comment]
> By the way I thought one of CIs runs such scenarios 🤷.

Unit tests - yes, benchmarks - no.
There is a separate benchmarking CI pocess that should be running manually before new release.

**Review by Alexey-N-Chernyshov — APPROVED** [review]
Good job! 🎉

**Pull request merged**